### PR TITLE
feat: Job holder usernames API

### PR DESCRIPTION
**Pull request by bilaltahir21**
**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/api/v1/views.py [hunk 1]
```diff
@@ -186,3 +187,38 @@ def get_queryset(self):
...
+                "user4",
+                "user5",
+                // 95 more usernames
+            ]
+        }
+        """
+        job = get_object_or_404(Job, id=job_id)
+
+        related_usernames = SkillsQuiz.objects.filter(current_job=job).order_by('-created')
+        usernames = related_usernames.order_by('-username').values_list('username', flat=True).distinct()[:100]
```
Comment (HammadAhmadWaqas) [inline taxonomy/api/v1/views.py:220]: Here we are adding two `oder_by`. Second Order_by is overriding first `order_by`. So the raw query for above query is followings
```
'SELECT DISTINCT `taxonomy_skillsquiz`.`username` FROM `taxonomy_skillsquiz` WHERE `taxonomy_skillsquiz`.`current_job_id` = 1 ORDER BY `taxonomy_skillsquiz`.`username` DESC LIMIT 100'
```

That clearly shows there is no order based on the `created` field. As field-based distinct is not supported in MySQL, We have to get distinct records using python. here is example query that I have tested and working good.
```
from collections import OrderedDict
usernames_query = SkillsQuiz.objects.filter(
    current_job_id=1
).order_by(
    '-id'
).values_list(
    "username", flat=True
)[:5000]
usernames = list(OrderedDict.fromkeys(usernames_query))[:100]
```

Comment (HammadAhmadWaqas) [reply taxonomy/api/v1/views.py:220]: Note: we can use -id as an alternative of -created, id ordering should be fast as it is a primary key.

**Review by HammadAhmadWaqas — CHANGES_REQUESTED** [review]
Can we please add unit tests for this API?

File: taxonomy/api/v1/views.py (hunk 1 above)
Comment (bilaltahir21) [reply taxonomy/api/v1/views.py]: @HammadAhmadWaqas I think this logic would not work in certain cases like if we have 5000 recent responses from the same user then would return just one user but that is highly unlikely. I have changed the usernames extraction logic as suggested.

File: taxonomy/api/v1/views.py [hunk 2]
```diff
@@ -186,3 +188,40 @@ def get_queryset(self):
...
+        }
+        """
+        job = get_object_or_404(Job, id=job_id)
+
+        usernames_queryset = SkillsQuiz.objects.filter(
+            current_job=job,
+        ).order_by('-id').values_list('username', flat=True)[:5000]
+        usernames = list(OrderedDict.fromkeys(usernames_queryset))[:100]
+
+        response_data = JobHolderUsernamesSerializer({'usernames': usernames}).data
```
Comment (HammadAhmadWaqas) [inline taxonomy/api/v1/views.py:225]: do we really need this `JobHolderUsernamesSerializer ` here? I think `usernames ` and `response_data ` is same. so we can simply `return Response(usernames)`

Comment (bilaltahir21) [reply taxonomy/api/v1/views.py]: Removed the serializer.

**Comment by bilaltahir21** [comment]
Output of the API with local data. 

User 1:
Screenshot 2022-11-22 at 5 42 51 PM

User 2: 
Screenshot 2022-11-22 at 5 41 55 PM

**Pull request merged**